src/merkle/merkle_tree.rs

```diff
@@ -24,7 +22,7 @@ impl MerkleTree {
     ///
     /// # Errors
     /// Returns an error if the number of leaves is smaller than two or is not a power of two.
-    pub fn new(leaves: Vec<Word>) -> Result<Self, MerkleError> {
+    pub fn new(leaves: Vec<RpoDigest>) -> Result<Self, MerkleError> {
```

Comment:
I'm wondering if we could keep the interface as it was (with `leaves` provided as `Word`'s) - but not sure how it would affect line 38.